grails-app/controllers/org/pih/warehouse/order/OrderController.groovy

```diff
@@ -354,8 +356,10 @@ class OrderController {
                 if (orderAdjustment.orderItem && !params.orderItem.id) {
                     orderAdjustment.orderItem.removeFromOrderAdjustments(orderAdjustment)
                 }
-                orderAdjustment.properties = params
                 def budgetCode = BudgetCode.get(params.budgetCode?.id)
+                params.remove("budgetCode.id")
+                params.remove("budgetCode")
+                orderAdjustment.properties = params
```

Comment:
Why do we need to do all this stuff with the params map? Line 362 above should be enough to handle budget code. Remove 359-361 and 363 and then let me know what happens.

Reply:
I made the changes myself in both places.

Reply:
@jmiranda It was a fix to the issue Katarzyna mentioned in the Jira ticket. I agree it was not pretty. I fixed it in a different way and pushed it here https://github.com/openboxes/openboxes/pull/1927